stellar-contract-env-host/src/host.rs

```diff
@@ -597,87 +603,143 @@ impl CheckedEnv for Host {
     }
 
     fn map_put(&self, m: Object, k: RawVal, v: RawVal) -> Result<Object, HostError> {
-        todo!()
+        let k = self.associate_raw_val(k);
```

Comment:
@jayz22 I missed you implemented these and I reimplemented them in #136. My impl was very very similar, so should be only a small conflict to fix if you merge the main branch into your PR. Sorry for the conflict.

Reply:
No worries @leighmcculloch . Will try to get this one merged soon. Currently still missing BigInt and unit tests.